dubbo-common/src/main/java/org/apache/dubbo/common/utils/NetUtils.java

```diff
@@ -139,13 +138,14 @@ static boolean isValidAddress(InetAddress address) {
         }
         String name = address.getHostAddress();
         return (name != null
-                && !ANYHOST.equals(name)
-                && !LOCALHOST.equals(name)
+                && !Constants.ANYHOST_VALUE.equals(name)
+                && !Constants.LOCALHOST_VALUE.equals(name)
                 && IP_PATTERN.matcher(name).matches());

```

Comment:
@CrazyHZM 
I think once the **name** is not null the rest of two evaulation 

```
!Constants.ANYHOST_VALUE.equals(name)
                && !Constants.LOCALHOST_VALUE.equals(name)
```

most likely be evaluates to **true** for non localhost so the chances of this whole condition to become true depends on **IP_PATTERN.matcher(name).matches()**.  So from better understanding and performance point of view I think it would be better if we change it to 

```
(name != null
                && IP_PATTERN.matcher(name).matches()
                && !Constants.ANYHOST_VALUE.equals(name)
                && !Constants.LOCALHOST_VALUE.equals(name));
```
What do yu say?

Reply:
thanks, I think this idea is good. I will modify it.